script.js

```diff
@@ -1,12 +1,200 @@
+  showSelectorContainer.innerText = '';
+  showSelectorContainer.appendChild(showSelector);
+}
+
+function selectShow(id) {
+  fetch(`https://api.tvmaze.com/shows/${id}/episodes`).then(resp => resp.json()).then(episodes => {
+    allEpisodes = episodes;
+
+    const root = document.getElementById("root");
+    root.innerText = '';
+
+    const backLink = document.createElement("a");
+    backLink.innerText = "< Select new show";
+    backLink.href = "javascript:setup()";
+    root.appendChild(backLink);
+
+    let episodesView = document.getElementById("episodes-view");
+    if (episodesView) {
+      episodesView.innerText = '';
+    } else {
+      episodesView = document.createElement("div");
+      episodesView.id = "episodes-view";
+      root.appendChild(episodesView);
+    }
+
+    const episodeSelector = document.createElement("select");
+    episodeSelector.id = "episode-selector";
+    episodeSelector.innerHTML = '<option value="">Select an episode</option>';
+    episodeSelector.addEventListener("change", selected);
+    episodesView.appendChild(episodeSelector);
+
+    const searchBox = document.createElement("input");
+    searchBox.type = "text";
+    searchBox.id = "search";
+    searchBox.placeholder = "Search by episode name";
+    searchBox.addEventListener("keyup", search)
+    episodesView.appendChild(searchBox);
+
+    const episodeCounter = document.createElement("span");
+    episodeCounter.id = "episode-count";
+    episodesView.appendChild(episodeCounter);
+
+    const episodesDisplay = document.createElement("div");
+    episodesDisplay.id = "episodesDisplay";
+    episodesView.appendChild(episodesDisplay);
+
+    render();
+  })
+}
+
+function selected(event) {
+  document.querySelectorAll(".episode-container.selected").forEach(element => {
+    element.className = element.className.replace(" selected ", " ")
+  })
+
+  render();
+}
+
+function highlight(element) {
+  element.className = `${element.className} selected `;
+}
+
+function search(event) {
+  render()
+}
+
+function zeroPad(number) {
+  if (number < 10) {
+    return `0${number}`;
+  }
+  return `${number}`;
+}
+
+function episodeCode(episode) {
+  return `S${zeroPad(episode.season)}E${zeroPad(episode.number)}`;
+}
+
+function render() {
+  window.scrollTo(0, 0);
+  const selector = document.querySelector("#episode-selector");
+  allEpisodes.forEach(episode => {
+    const option = document.createElement("option");
+    const code = episodeCode(episode);
+    option.value = code;
+    option.text = `${code} - ${episode.name}`;
+    selector.appendChild(option)
+  });
+
+  const selectedCode = document.getElementById("episode-selector").value;
+  const searchTerm = document.getElementById("search").value.toLowerCase();
+  const episodeList = allEpisodes
+    .filter(episode => !selectedCode || episodeCode(episode) == selectedCode)
+    .filter(episode => episode.name.toLowerCase().includes(searchTerm));
```

Comment:
The search requirement is that matches can also be found in the episode summary, not only in the episode name.

Reply:
Oops! Good catch! Fixed - and even extracted a named function for it!